arelle/ModelValue.py

```diff
@@ -122,80 +125,63 @@ def qnameEltPfxName(element, prefixedName, prefixException=None):
-        except AttributeError:
-            return False
-        ''' don't think this is used any longer
-        if isinstance(other,_STR_BASE):
-            # only compare nsnames {namespace}localname format, if other has same hash
-            return self.__hash__() == other.__hash__() and self.clarkNotation == other
-        elif isinstance(other,QName):
-            return self.qnameValueHash == other.qnameValueHash and \
-                    self.namespaceURI == other.namespaceURI and self.localName == other.localName
-        elif isinstance(other,ModelObject):
-            return self.namespaceURI == other.namespaceURI and self.localName == other.localName
-        '''
-        '''
-        try:
-            return (self.qnameValueHash == other.qnameValueHash and 
-                    self.namespaceURI == other.namespaceURI and self.localName == other.localName)
-        except AttributeError:  # other may be a model object and not a QName
-            try:
-                return self.namespaceURI == other.namespaceURI and self.localName == other.localName
-            except AttributeError:
-                return False
+
+    def __eq__(self, other):
+        if QName.is_instance(other):
+            return self.localName == other.localName and self.namespaceURI == other.namespaceURI
         return False
-        '''
-    def __ne__(self,other):
-        return not self.__eq__(other)
-    def __lt__(self,other):
-        return (self.namespaceURI is None and other.namespaceURI) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI < other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName < other.localName)
-    def __le__(self,other):
-        return (self.namespaceURI is None and other.namespaceURI) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI < other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName <= other.localName)
-    def __gt__(self,other):
-        return (self.namespaceURI and other.namespaceURI is None) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI > other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName > other.localName)
-    def __ge__(self,other):
-        return (self.namespaceURI and other.namespaceURI is None) or \
-                (self.namespaceURI and other.namespaceURI and self.namespaceURI > other.namespaceURI) or \
-                (self.namespaceURI == other.namespaceURI and self.localName >= other.localName)
+
+    def __lt__(self, other):
+        if QName.is_instance(other):
+            return (_default_to_empty(self.namespaceURI), _default_to_empty(self.localName)) < (_default_to_empty(other.namespaceURI), _default_to_empty(other.localName))
+        raise NotImplementedError(f'Comparison between {type(self)} and {type(other)} not supported')
```

Comment:
Let the interpreter attempt reversing the operands and throw a proper type error if that also doesn't work.
```suggestion
        return NotImplemented
```